prow/git/v2/client_factory.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+// ClientFactory knows how to create clientFactory for repos
+type ClientFactory interface {
+	// ClientFromDir creates a client that operates on a repo that has already
+	// been cloned to the given directory.
+	ClientFromDir(org, repo, dir string) (RepoClient, error)
+	// ClientFor creates a client that operates on a new clone of the repo.
+	ClientFor(org, repo string) (RepoClient, error)
```

Comment:
nit: Maybe mention that this client uses a local clone cache? That might be useful to know since it speeds up clones of the same repo and increases disk usage.

Reply:
Also clarify that this is for standard GitHub repos and provide a variant for GitHub enterprise repos that need to specify a different clone URL.

Reply:
> Also clarify that this is for standard GitHub repos and provide a variant for GitHub enterprise repos that need to specify a different clone URL.

I think that should be configured on the `ClientFactory` impementation. Or do we have to build this in  a way that it works for both GitHub and GHE simultanously?